mount-partitions.sh

```diff
@@ -30,14 +30,58 @@ dbus_get_mount() {
     fi
 }
 
+# Identify an encrypted partition using dmsetup
+# Works for both dm-crypt and LUKS encrypted partitions
+# Arguments:
+# 1: Partition device - will be converted to a DM device
+# Returns:
+# 0: The partition device is encrypted
+# 1: The partition device is not encrypted
+is_part_encrypted() {
+	_part="${1#/dev/}"
+	_dm_part="${_part}"
+	if command -v dmsetup > /dev/null; then
+		if [ "${_part#dm-}" = "${_part}" ]; then
+			# Does not start with dm-
+			if [ "${_part#mapper/}" = "${_part}" ]; then
+				# Does not start with mapper/
+				# Find the corresponding DM device to the partition
+				_dm_part=$(lsblk -nlo kname "/dev/${_dm_part}" | grep dm)
+				if [ -z "${_dm_part}" ]; then
+					# No corresponding DM device, no dm-crypt in use
+					return 1
+				fi
+			fi
+		fi
+		# _dm_part is a DM device, either dm* or mapper/*
```

Comment:
This point will be reached either if 

- `_part` starts with `dm-`
- `_part` starts with `mapper/`
-  `lsblk -nlo kname "/dev/${_dm_part}" | grep dm` returns something

Is that the logic that is intended? (I'm not familiar with the partition layout in NXP devices)

Reply:
@pipex if the device starts with `dm-` or `mapper` it is a DM device, if not, if falls back to listing the kernel device name and makes sure it does not contain `dm` before discarding a DM device.
If it is a DM device, then it will use its name to filter the `dmsetup` list of dm-crypt devices.
In all, it tries hard to identify non-dm devices and confirm encrypted dm devices.

Reply:
I see, I added one suggestion above, otherwise it looks good to me. I cannot, however provide any suggestions relating to the used tools or the NXP context